syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -812,3 +813,28 @@ def combine(self, *pointers):
         ps.append(self)
 
         return syft.combine_pointers(*ps)
+
+    def encrypt(self, public_key):
+        """This method will encrypt each value in the tensor using Paillier
+        homomorphic encryption.
+
+        Args:
+            *public_key a public key created using
```

Comment:
docstring style Google

Reply:
I was wondering if `encrypt` shoudln't be the entry point both for HE and MPC, you'd have a method argument which is equal to "he" or "mpc" and depending on which you call the he logic you wrote or the .share (and possibly the .fix_prec as well) ; you'd add the other options for encryption (public_key, or workers, or cryptoprovider also)

Reply:
I dunno - I like the distinction between encrypting something and sharing something. If I'm reading code, it's a lot easier to read the difference between the words ".share() and .encrypt() than .encrypt(bobs_key, proto=he) and .encrypt(bobs_worker, alices_worker, proto=mpc). The first you know immediately the purpose fo the function. The second you don't find out until the end (classic example for an increase in cognitive load)